openlibrary/plugins/upstream/account.py

```diff
@@ -402,6 +403,43 @@ def GET(self):
         # Show failed message without thinking.
         return render['account/verify/failed']()
 
+class account_validation(delegate.page):
+    path = '/account/validate'
+
+    @staticmethod
+    def validate_username(username):
+        if not 3 <= len(username) <= 20:
+            return _('Username must be between 3-20 characters')
+        if not re.match('^[A-Za-z0-9-_]{3,20}$', username):
+            return _('Username may only contain numbers and letters')
+        ol_account = OpenLibraryAccount.get(username=username)
+        if ol_account:
+            return _("Username unavailable")
+
+    @staticmethod
+    def validate_email(email):
+        if not (email and re.match('.*@.*\..*', email)):
+            return _('Must be a valid email address')
+
+        ol_account = OpenLibraryAccount.get(email=email)
+        if ol_account:
+            return _('Email already registered')
+
+
+    def GET(self):
+        i = web.input()
+        errors = {
+            'email': None,
+            'username': None
+        }
+        if i.get('email') is not None:
+            errors['email'] = self.validate_email(unquote_plus(i.email))
+        if i.get('username') is not None:
```

Comment:
```suggestion
        if i.get('email'):
            errors['email'] = self.validate_email(unquote_plus(i.email))
        if i.get('username'):
```
Deal properly with all _falsey_ such as None, "", {}, [], (,), False...
In the near future we may swap out web.py for more a modern frame that may return slightly different output.

Reply:
In this specific case, if e.g. an empty string `''` is provided, we want to enter this block because the API is being passed an empty email. 

e.g. https://dev.openlibrary.org/account/validate?email= should return invalid email